src/golibmc.go

```diff
@@ -148,15 +181,110 @@ use in every retrieval/storage command. default False.
 */
 func New(servers []string, noreply bool, prefix string, hashFunc int, failover bool, disableLock bool) (client *Client) {
 	client = new(Client)
-	client._imp = C.client_create()
-	runtime.SetFinalizer(client, finalizer)
+	client.servers = servers
+	client.prefix = prefix
+	client.noreply = noreply
+	client.disableLock = disableLock
+	client.hashFunc = hashFunc
+	client.failover = failover
+	client.openerCh = make(chan struct{}, connectionRequestQueueSize)
+	client.connRequests = make(map[uint64]chan *conn)
+	client.maxOpen = 1 // default value
+
+	go client.opener()
+
+	return
+}
+
+// SimpleNew to create a memcached client with default params
+func SimpleNew(servers []string) (client *Client) {
+	noreply := false
+	prefix := ""
+	hashFunc := HashCRC32
+	failover := false
+	disableLock := false
+	return New(
+		servers, noreply, prefix, hashFunc,
+		failover, disableLock,
+	)
+}
+
+func finalizer(cn *conn) {
+	C.client_destroy(cn._imp)
+}
+
+func (client *Client) lock() {
+	if !client.disableLock {
+		client.lk.Lock()
+	}
+}
+
+func (client *Client) unlock() {
+	if !client.disableLock {
+		client.lk.Unlock()
+	}
+}
+
+func (client *Client) maybeOpenNewConnections() {
+	if client.closed {
+		return
+	}
+	numRequests := len(client.connRequests)
+	if client.maxOpen > 0 {
+		numCanOpen := client.maxOpen - client.numOpen
+		if numRequests > numCanOpen {
+			numRequests = numCanOpen
+		}
+	}
+	for numRequests > 0 {
+		client.numOpen++
+		numRequests--
+		client.openerCh <- struct{}{}
+	}
+}
 
-	n := len(servers)
+func (client *Client) opener() {
+	for range client.openerCh {
+		client.openNewConnection()
+	}
+}
+
+func (client *Client) openNewConnection() {
+	if client.closed {
+		client.lk.Lock()
+		client.numOpen--
+		client.lk.Unlock()
+		return
+	}
+	cn, err := client.newConn()
+	if err != nil {
+		client.lk.Lock()
+		defer client.lk.Unlock()
+		client.numOpen--
+		client.maybeOpenNewConnections()
+		return
+	}
+	client.lk.Lock()
+	defer client.lk.Unlock()
+	if !client.putConnLocked(cn) {
+		client.numOpen--
```

Comment:
also need to close cn here

Reply:
Fixed.
https://github.com/douban/libmc/pull/57/commits/d73a6d1b46103bb735716cf7edae9557a93465f2

Reply:
https://github.com/douban/libmc/pull/57/files#diff-b21c9a32b7563e26c49d7f215bc5a656R274 can't see the fix?

Reply:
just ping

Reply:
Sorry I didn't notice this. Fixed.
https://github.com/douban/libmc/pull/57/commits/f038ae6c95d45929c4eec1a1821a43c851a48318